### PR TITLE
[E2E] forward exit code to `host.Execute`

### PR DESCRIPTION
**Pull request by clarkb7**
### What does this PR do?
Use `exit $LASTEXITCODE` to set the PowerShell/command exit code when running commands with `host.Execute`.

Removes the env var clearing after the command. Each `host.Execute` call creates a new PowerShell session, so there will be no leakage or cross contamination between commands.

### Motivation
E2E tests should be able to examine the exit code of commands.

Continuation of https://github.com/DataDog/datadog-agent/pull/28291, which had a similar effect only for PowerShell cmdlets.

By default PowerShell only returns 0 or 1, you have to call `exit` for a custom error code.

The env var reset commands were hiding errors. Since they run after the caller provided command, they reset/clear the error state, causing `host.Execute` to always return success for commands run with env vars.

### Describe how to test/QA your changes
N/A, test change
testcase added
```
go test -v ./examples --run TestVMSuiteWithFileOperations/TestCommandResults
```

### Possible Drawbacks / Trade-offs
Test case added to `examples` folder. is there a better place for them (and the file operation tests)?

Does not add detailed test cases for Linux platforms

### CODE REVIEW

**Comment by clarkb7** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by clarkb7** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

File: test/new-e2e/examples/vm_with_file_operations_test.go [hunk 1]
```diff
@@ -22,10 +25,77 @@ type vmSuiteWithFileOperations struct {
...
+	v.Assert().Contains(out, "hello")
+
+	// invalid commands should return an error
+	_, err = vm.Execute("not-a-command")
+	v.Assert().Error(err, "invalid command should return an error")
+
+	// specific exit code should be returned
+	_, err = vm.Execute("exit 2")
+	v.Assert().Error(err, "non-zero exit code should return an error")
+	assertExitCodeEqual(v.T(), err, 2, "specific exit code should be returned")
```
Comment (pducolin) [inline test/new-e2e/examples/vm_with_file_operations_test.go:55]: **🥜 nitpick**
Would rather have the assertion explicit, as in 
```suggestion
	assert.Equal(t, expected, exitErr.ExitStatus(), msgAndArgs)
```

or move the helper closer to `host.go`. As is I consider it hard to discover that the error returned by `Execute` is an `ssh.Error`. Should we have `ExecuteWithExitCode` instead, or `Execute` returning `output, exitCode, error`?

Comment (pducolin) [reply test/new-e2e/examples/vm_with_file_operations_test.go:55]: I would keep everything here - I know golang is sometimes verbose, but this helps readability in this example:
```suggestion
	var exitErr *ssh.ExitError
	require.ErrorAs(v.T(), err, &exitErr, "error code should contain ssh.ExitError in its chain")
	assert.Equal(v.T(), exitErr.ExitStatus(), 2, "specific exit code should be returned")
```

File: test/new-e2e/pkg/utils/e2e/client/host.go [hunk 2]
```diff
@@ -516,11 +517,23 @@ func buildCommandOnWindows(h *Host, command string, envVar EnvVar) string {
 
 		cmd += fmt.Sprintf("$env:%s='%s'; ", envName, envValue)
 	}
-	cmd += fmt.Sprintf("%s; ", command)
-
-	for envName := range envVar {
-		cmd += fmt.Sprintf("$env:%s='%s'; ", envName, envVarSave[envName])
-	}
```
Comment (pducolin) [inline test/new-e2e/pkg/utils/e2e/client/host.go:523]: **❓ question**
Was this removed on purpose? we were adding them to reset the env vars to previous values after command execution, is it not required on PowerShell?

**Review by pducolin — APPROVED** [review]

**Comment by pducolin** [comment]
Thank you for the fix, we should have this test added to `test/new-e2e/test-infra-definiton` that is currently executed on CI in `pre_e2e` as integration tests for the framework.

File: test/new-e2e/pkg/utils/e2e/client/host.go (hunk 2 above)
Comment (clarkb7) [reply test/new-e2e/pkg/utils/e2e/client/host.go:523]: yes, it was the cause of the bug. See PR description and code comment for details.

**Pull request merged**